_posts/2024-02-01-whats-new-in-provisioning.adoc

```diff
@@ -0,0 +1,94 @@
+
+* Part of the content (the set of bash scripts that are run at server startup) packaged 
+in the `org.wildfly.cloud:wildfly-cloud-galleon-pack` that is used for cloud context is not usable when the server is packaged as a Bootable JAR.
+
+* In a cloud context, the kind of packaging is not really relevant. WildFly servers are not directly started. 
+Containers are started whatever the kind of installed WildFly server.
+
+So the JAR produced by the WildFly Maven Plugin is not expected to be used in a cloud context.
+
+For cloud we advise to provision a WildFly server. To do so:
+
+* Use WildFly Glow support (`<context>cloud</context>`) that will automatically retrieve the required Galleon feature-packs or 
+* if not using WildFly Glow, add the `org.wildfly.cloud:wildfly-cloud-galleon-pack` feature-pack to the `feature-packs` list.
+
+== Galleon Upward compatibility
+
+WildFly provisioning tooling heavily relies on https://docs.wildfly.org/galleon/[Galleon] technology.
+
+Galleon uses the WildFly feature-packs to provision a WildFly server.
+
+One issue that we have been facing in the past is that, when the structural content of a feature-pack changes 
+(due to an evolution of Galleon core library or new XML schema), the provisioning tooling must be upgraded to be able to handle the new format. 
+The failure occurs during provisioning printing an hard to decrypt error message.
+
+The root cause of this problem is that a WildFly feature-pack can require a specific version of Galleon core library to be properly provisioned, 
+and this version can differ from the version that the provisioning tooling (https://docs.wildfly.org/galleon/[Galleon CLI or Maven Plugin], https://github.com/wildfly/wildfly-maven-plugin[WildFly Maven Plugin], https://github.com/wildfly-extras/wildfly-jar-maven-plugin[WildFly Bootable JAR Maven Plugin], https://github.com/wildfly/wildfly-glow[WildFly Glow]) 
+directly depends on.
+
+To solve this problem we have evolved Galleon.
+
+=== WildFly Galleon feature-pack build time
+
+When using https://docs.wildfly.org/galleon-plugins/[WildFly Galleon Maven Plugin] 7.0.0.Beta1, the Galleon version used to build the feature-pack is included in the feature-pack. 
+A check is done at build time to validate that the content of the feature-pack is valid with respect to the Galleon version in use.
+If some content is not supported, the build fails, and an upgrade of the WildFly Galleon Maven Plugin is required.
+ 
+=== WildFly provisioning time
+
+We have been introducing a new https://github.com/wildfly/galleon/tree/main/api[Galleon tooling API] that isolates the provisioning tooling 
+from the Galleon Core version. The API implementation detects the core version required by the feature-packs and uses it.
+
+So we should now be protected at provisioning time from incompatible changes. This API is backward compatible with older WildFly feature-packs.
+
+The following provisioning tooling have been ported to the new API:
```

Comment:
tooling is singular so 'tooling have' sounds a bit odd. So does 'toolings have' and I think 'tooling has' (since there are more than one!)

Maybe 'tools have' or 'tooling libraries have'?